tests/driver_at86rf2xx/main.c

```diff
@@ -93,11 +93,13 @@ int main(void)
 
     for (unsigned i = 0; i < AT86RF2XX_NUM; i++) {
         const at86rf2xx_params_t *p = &at86rf2xx_params[i];
+        netdev2_t *dev = (netdev2_t *)(&devs[i]);
+
         printf("Initializing AT86RF2xx radio at SPI_%d\n", p->spi);
         at86rf2xx_setup(&devs[i], p->spi, p->spi_speed, p->cs_pin,
                         p->int_pin, p->sleep_pin, p->reset_pin);
-        devs[i].driver->init((netdev2_t *)&devs[i]);
-        devs[i].event_callback = _event_cb;
+        dev->driver->init(dev);
```

Comment:
I'd say the hard fault would not happen due to checks like https://github.com/RIOT-OS/RIOT/pull/4646/files#diff-8f19223a80f38940500cf05c0e76cde5R216, but it definitely makes more sense if the call back is set before initialization. Will fix.